SEO-477 Rebuild Code for Legacy Category (to update the Pagination)

gabrys opened the pull request.

Commit:
SEO-477 Remove category page tweak fors blogs

Commit:
SEO-477 Category page rewrite

WikiaTech commented on extensions/wikia/CategoryPagination/templates/category.tmpl.php:
```diff
@@ -0,0 +1,36 @@
+	<?= $categoryGallery; ?>
+
+	<div lang="<?= htmlspecialchars( $lang->getCode() ); ?> dir="<?= htmlspecialchars( $lang->getDir() ); ?>">
+	<? if ( $content['subcat'] ): ?>
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "subcat" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
-			'file' => $cat->getFileCount(),
-			'subcat' => $cat->getSubcatCount(),
-			'page' => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
+			self::TYPE_FILE => $cat->getFileCount(),
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
-			'subcat' => $cat->getSubcatCount(),
-			'page' => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
+			self::TYPE_FILE => $cat->getFileCount(),
+			self::TYPE_SUBCAT => $cat->getSubcatCount(),
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
-			'page' => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
+			self::TYPE_FILE => $cat->getFileCount(),
+			self::TYPE_SUBCAT => $cat->getSubcatCount(),
+			self::TYPE_PAGE => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
+		} );
+
+		// Get subcategories
+		$this->processSection( self::TYPE_SUBCAT, function ( $title, $humanSortkey, $row ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
+		$url = $cat->getTitle()->getFullURL();
+		$this->page = max( 1, $this->getRequest()->getInt( 'page', 1 ) );
+		foreach ( array_keys( $this->counts ) as $type ) {
+			// TODO: add the #mw-pages, #mw-category-media and #mw-subcategories fragments
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 36 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 17 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/64a4d11eaa5632c8ef51358fc6ec13e6ab4b767b/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
+			}
+		}
+		if ( $maxPaginator ) {
+			// TODO: make sure to link without the #mw-pages, #mw-category-media and #mw-subcategories fragments
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
+
+		$dbr = wfGetDB( DB_SLAVE, 'category' );
+		$result = $dbr->select(
 			[ 'page', 'categorylinks' ],
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "categorylinks" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
+		} );
+
+		// Get media
+		$this->processSection( self::TYPE_FILE, function ( $title, $humanSortkey, $row ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,221 @@
-				$numPages = $paginator->getPagesCount();
-				$maxPaginator = $paginator;
+		// Get pages
+		$this->processSection( self::TYPE_PAGE, function ( $title, $humanSortkey, $row ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 36 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 17 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/c049c65743f402f846b920b82b83f8f8c95b1613/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented:
SonarQube analysis reported 57 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 36 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 17 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 25 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/2c2bc2e40ff74c03af95496efa646ca14f9a3638/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

mszabo-wikia commented on extensions/wikia/CategoryPagination/templates/category.tmpl.php:
```diff
@@ -0,0 +1,36 @@
+<? if ( $empty ): ?>
+	<?= wfMsgExt( 'category-empty', array( 'parse' ) ); ?>
```
Should be `wfMessage` throughout

saipetch commented on extensions/wikia/Blogs/BlogArticle.php:
```diff
@@ -348,75 +348,6 @@ static public function getProps( $page_id ) {
-
-		$dbcnt = self::blogsInCategory( $cat );
-		$rescnt = count( $catView->blogs );
-		$countmsg = self::getCountMessage( $catView, $rescnt, $dbcnt, 'article' );
```
consider removing getCountMessage() function ?

gabrys replied:
done

saipetch commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,230 @@
+		wfRunHooks( 'CategoryViewer::beforeCategoryData', array( &$conds ) );
+
+		$dbr = wfGetDB( DB_SLAVE, 'category' );
+		$result = $dbr->select(
```
consider using WikiaSQL() ?

gabrys replied:
Wall extension relies on the SQL to be constructed using $where array, so it's easier to keep as is.

Commit:
SEO-477 Remove unused function from BlogArticle class

WikiaTech commented:
SonarQube analysis reported 56 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 35 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 17 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/fceb4b096c21683908779048e8318a3e974da2c4/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

Commit:
SEO-477 Converted messages in category.tmpl.php to the new API

WikiaTech commented on extensions/wikia/CategoryPagination/templates/category.tmpl.php:
```diff
@@ -0,0 +1,36 @@
+	<?= $categoryGallery; ?>
+
+	<div lang="<?= htmlspecialchars( $lang->getCode() ); ?> dir="<?= htmlspecialchars( $lang->getDir() ); ?>">
+	<? if ( !empty( $content['subcat'] ) ): ?>
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "subcat" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented:
SonarQube analysis reported 56 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 35 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 17 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/d46f246febcc0744dd5777464b58dd102ca4dab3/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

Commit:
SEO-477 Renamed a Paginator method to get the current page date

WikiaTech commented:
SonarQube analysis reported 64 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 36 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 23 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/bb3143ddb0e25f2d607520e31884e7f5ba3eef8f/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

saipetch commented on extensions/wikia/CategoryPagination/templates/category.tmpl.php:
```diff
@@ -0,0 +1,36 @@
+<? if ( $empty ): ?>
+	<?= wfMessage( 'category-empty' )->parse(); ?>
+<? else: ?>
+	<?= $categoryGallery; ?>
```
Do we want to put $categoryGallery above main div ?

saipetch commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,230 @@
-			$humanSortKey = $title->getCategorySortkey( $row->cl_sortkey_prefix );
-			return $humanSortKey;
+		$numberShown = 0;
+		if ( $result->numRows() ) {
```
consider removing if-clause ?

saipetch commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,230 @@
 				'ORDER BY' => 'cl_sortkey',
+				'LIMIT' => $this->limit,
+				'OFFSET' => $this->limit * ( $this->paginators[$type]->getActivePage() - 1 ),
+				'USE INDEX' => array( 'categorylinks' => 'cl_sortkey' ),
```
'USE INDEX' => [ 'categorylinks' => 'cl_sortkey' ], ?

saipetch commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,230 @@
+		// Set up pagination
+		$url = $cat->getTitle()->getFullURL();
+		$this->page = max( 1, $this->getRequest()->getInt( 'page', 1 ) );
+		foreach ( array_keys( $this->counts ) as $type ) {
```
foreach ($this->counts as $type => $value) {
    $this->paginators[$type] = new Paginator( $value, $this->limit, $url );
    $this->paginators[$type]->setActivePage( $this->page );
}
?

Commit:
SEO-477 Code review comments addressed

WikiaTech commented on extensions/wikia/CategoryPagination/templates/category.tmpl.php:
```diff
@@ -0,0 +1,36 @@
+	<div lang="<?= htmlspecialchars( $lang->getCode() ); ?> dir="<?= htmlspecialchars( $lang->getDir() ); ?>">
+	<?= $categoryGallery; ?>
+
+	<? if ( !empty( $content['subcat'] ) ): ?>
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "subcat" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech replied:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "subcat" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,226 @@
-			'file' => $cat->getFileCount(),
-			'subcat' => $cat->getSubcatCount(),
-			'page' => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
+			self::TYPE_FILE => $cat->getFileCount(),
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech replied:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,226 @@
-			'subcat' => $cat->getSubcatCount(),
-			'page' => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
+			self::TYPE_FILE => $cat->getFileCount(),
+			self::TYPE_SUBCAT => $cat->getSubcatCount(),
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech replied:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,226 @@
-			'page' => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
+			self::TYPE_FILE => $cat->getFileCount(),
+			self::TYPE_SUBCAT => $cat->getSubcatCount(),
+			self::TYPE_PAGE => $cat->getPageCount() - $cat->getFileCount() - $cat->getSubcatCount(),
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech replied:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented:
SonarQube analysis reported 62 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 36 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 23 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/5ef2180c161bd48bc1ae137788caecd0c3a12c7a/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

Commit:
SEO-477 Call Paginator::getBarHTML by the proper method name

Method and function names in PHP are not case sensitive, but the good
practice is to call the methods and functions like they were case
sensitive to avoid confusion and false negatives when searching for
method usages.

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,226 @@
+		} );
+
+		// Get media
+		$this->processSection( self::TYPE_FILE, function ( $title, $humanSortkey, $row ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,226 @@
+
+		$dbr = wfGetDB( DB_SLAVE, 'category' );
+		$result = $dbr->select(
 			[ 'page', 'categorylinks' ],
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "categorylinks" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,226 @@
-				$numPages = $paginator->getPagesCount();
-				$maxPaginator = $paginator;
+		// Get pages
+		$this->processSection( self::TYPE_PAGE, function ( $title, $humanSortkey, $row ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented:
SonarQube analysis reported 62 issues
- ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 36 major
- ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 23 minor
- ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L14](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L14): Class "BlogArticle" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L64](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L64): Remove this unused "$author" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
3. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L69](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L69): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
4. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L130](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L130): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
5. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L197](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L197): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
6. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L206](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L206): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
7. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L216](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L216): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
8. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L248](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L248): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
9. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L318](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L318): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
10. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [BlogArticle.php#L339](https://github.com/Wikia/app/blob/522c452ee5d6e8315845fb06a9c3850b61810da4/extensions/wikia/Blogs/BlogArticle.php#L339): Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

WikiaTech commented on extensions/wikia/CategoryPagination/CategoryPaginationViewer.class.php:
```diff
@@ -2,102 +2,226 @@
+		} );
+
+		// Get subcategories
+		$this->processSection( self::TYPE_SUBCAT, function ( $title, $humanSortkey, $row ) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

saipetch commented:
LGTM
